### PR TITLE
10 redirect to login before signin

### LINKED ISSUE #10: Redirect to /login before signin

### PR DESCRIPTION
**Pull request by wood-crafter**

### CODE REVIEW

**Comment by wood-crafter** [comment]
@hoangph271 
redirect not working when i using path /home and /about.
Please investigate and show me how to solve

File: src/views/about/index.js [hunk 1]
```diff
@@ -1,12 +1,22 @@
...
 export const About = () => {
-  return (
-    <div>
-      <NavBar />
-      <h4>
-        About...!
-      </h4>
-    </div>
-  )
+  const { user } = useContext(AuthContext)
```
Comment (hoangph271) [inline src/views/about/index.js:7]: Prefer this pattern...!

```jsx
import { Navigate } from "react-router-dom";

// This is a *guard return*, it deals with null data
if (!user) return <Navigate to="/login" />
// Notice I use <Navigate /> - This make our function return only one type of data: JSX

// Just return normally
return (
  // About JSX here
)
```

File: src/views/home/index.js [hunk 2]
```diff
@@ -1,16 +1,20 @@
 import { NavBar } from '../../components'
-import React, { useEffect, useState } from 'react'
+import React, { useEffect, useState, useContext } from 'react'
+import { AuthContext } from '../../providers/use-auth'
+import { useNavigate } from 'react-router'
 import './home.css'
 
 const getDog = async () => {
```
Comment (hoangph271) [inline src/views/home/index.js:7]: I'm not sure about this part, looks like `getDog()` and other dog related stuff shouldn't be in `index.js`.  

But I'll give you a pass on this...!  
Maybe create an issue to refactor this...?

**Review by hoangph271 — CHANGES_REQUESTED** [review]

**Comment by hoangph271** [comment]
> @hoangph271 redirect not working when i using path /home and /about. Please investigate and show me how to solve

Try fixing my comments and see if it helps with this issue...?

File: src/views/home/index.js (hunk 2 above)
Comment (wood-crafter) [reply src/views/home/index.js:7]: I got it, thank you

File: src/views/about/index.js (hunk 1 above)
Comment (wood-crafter) [reply src/views/about/index.js:7]: I resolved, but still dont know why it works @hoangph271

**Review by hoangph271 — APPROVED** [review]

File: src/views/about/index.js (hunk 1 above)
Comment (hoangph271) [reply src/views/about/index.js:7]: > It's pretty well explained in this browser warning...!

image

A React `FunctionalComponent` (`FC`) should always return JSX.  
And other logic should be handled in `useEffect` or other hooks.  
Don't mess with ***render logic*** - this means the return part of the `FC`

**Comment by hoangph271** [comment]
@wood-crafter btw try to come up with better commit messages

- What you want to have should be the branch name, eg: `feature/login_page`, `fix/sign_up_validation`  
- What you did should be used to write the commit message: It explains how the code is changed:
```
[Fix] Use <Navigate /> instead of navigate()
[Add] Added Login page
```

**Pull request merged**